pysidt/sidt.py

```diff
@@ -1102,7 +1104,20 @@ def estimate_uncertainty(self,rel_node_dof_tolerance=1e-5):
                 node.rule.uncertainty = node.parent.rule.uncertainty
             elif node.rule.num_data == 0:
                 node.rule.uncertainty = 0.0 #if n=0 the LASSO should drive node.rule.value to zero so there should be approximately no variance contribution 
-
+                
+        if self.validation_set and len(node_uncertainties) > 0:
```

Comment:
Enable assignment of the validation_set in this function in case someone wants to tune it after generating the tree easily.